lib/OpenQA/WebAPI/Controller/Running.pm

```diff
@@ -115,9 +114,9 @@ sub livelog {
     # logfile and send them to the client, plus a utility function to
     # close the connection if anything goes wrong.
     my $id;
-    my $close = sub {
+    local *doclose = sub {
```

Comment:
Remember that during the enclosing scope, a dynamically bound function is still available from the outside. I doubt anything would try to call it, but you don't know.